shared/actions/config.js

```diff
@@ -142,7 +143,8 @@ const bootstrap = (opts: $PropertyType<ConfigGen.BootstrapPayload, 'payload'>):
           logger.flush()
         })
         dispatch(NotificationsGen.createListenForKBFSNotifications())
-        if (!opts.isReconnect) {
+        if (!didInitialNav) {
```

Comment:
I don't understand what the logic is in general. We need to test this change a bunch

Reply:
Sure, I agree re. testing this a bunch.

My understanding here is that bootstrap is called both at startup and whenever reachability changes to 'reachable'. We want to do the initial navigation only the first time bootstrap is run, so that dropping the network and regaining it doesn't change the page you're on. The bug here was that we were assuming that all bootstraps coming from the reachability change (i.e. with `opts.isReconnect` set to true) are after the initial bootstrap, but in fact the initial bootstrap can fail.